### DISCUSSION TITLE
BelongsTo field with dependsOn retains old selected value when changing parent (NOT SOLVED)

### DISCUSSION THREAD

**Discussion by emcarson** [laravel/nova-issues#5677] [Q&A]
The reported issue: https://github.com/laravel/nova-issues/issues/5674
Was **NOT** solved.

First, you affirm that is the intended behaviour of a BelongsTo field is to retain an invalid value even though the options have changed, offering no explanation as to why or what is the possible use case for this.

Second, you offer a "solution" which I test and let you know that isn't working.

Your answer? Closing the issue as "not planned". 

What's not planned? Fixing a bug?
Or giving a solution to a real, demostrable problem to a paying customer?

Frankly I'm appalled by the customer support of this closed source, commercial product.

**Comment by crynobone** [comment]
> offering no explanation as to why or what is the possible use case for this.

The explanation is it is the expected behavior. The underlying code consists of a complex conditional structure with multiple possibilities that we can't 100% rely on options and seletedResourceId will be synced in all scenario. Reading the code under `resources/js/fields/Form/BelongsToField.vue` and `resources/mixins/DependentFormField.js` is a good start.

The current validation error should already be a good indication that the user needs to correct the value to proceed, then closing the issue without further action.

**Reply by emcarson** [reply]
> The current validation error should already be a good indication that the user needs to correct the value to proceed

And what if the second belongsTo field is nullable? 

Should the user select an option even though he's allowed not to, just to get past the validation error?
Or should he refresh the page, losing all progress just to start typing again?

**Reply by crynobone** [reply]
Submit a bug report with that exact example with a full reproducing repository as suggested in the Bug report template.

**Reply by emcarson** [reply]
Nah, I lost enough time already, I won't spend another couple of hours setting that up just for it to get archived/ignored.
More so if we're talking of "expected behaviour".
